integrations/oci/metrics/metrics/src/main/java/io/helidon/integrations/oci/metrics/OciMetricsSupport.java

```diff
@@ -0,0 +1,527 @@
+            return;
+        }
+
+        if (scopes.length == 0) {
+            LOGGER.info("No selected metric scopes to push to OCI");
+            return;
+        }
+
+        LOGGER.fine("Starting OCI Metrics agent");
+
+        RegistryFactory rf = RegistryFactory.getInstance();
+        Stream.of(scopes)
+                .forEach(type -> metricRegistries.put(rf.getRegistry(type), type));
+
+        ociMetricsData = new OciMetricsData(
+                metricRegistries, nameFormatter, compartmentId, namespace, resourceGroup, descriptionEnabled);
+        startExecutor();
+    }
+
+    private void prepareShutdown(WebServer webServer) {
+        webServerCounter.incrementAndGet();
+        webServer.whenShutdown().thenAccept(this::shutdown);
+    }
+
+    /**
+     * Fluent API builder to create {@link OciMetricsSupport}.
+     */
+    @Configured
+    public static class Builder implements io.helidon.common.Builder<Builder, OciMetricsSupport> {
+
+        private static final long DEFAULT_SCHEDULER_INITIAL_DELAY = 1L;
+        private static final long DEFAULT_SCHEDULER_DELAY = 60L;
+        private static final long DEFAULT_BATCH_DELAY = 1L;
+        private static final TimeUnit DEFAULT_SCHEDULER_TIME_UNIT = TimeUnit.SECONDS;
+        private static final int DEFAULT_BATCH_SIZE = 50;
+
+        private static final Map<String, Type> SCOPE_TYPES = Map.of(
+                Type.BASE.getName(), Type.BASE,
+                Type.VENDOR.getName(), Type.VENDOR,
+                Type.APPLICATION.getName(), Type.APPLICATION
+        );
+
+        private long initialDelay = DEFAULT_SCHEDULER_INITIAL_DELAY;
+        private long delay = DEFAULT_SCHEDULER_DELAY;
+        private long batchDelay = DEFAULT_BATCH_DELAY;
+        private TimeUnit schedulingTimeUnit = DEFAULT_SCHEDULER_TIME_UNIT;
+        private String compartmentId;
+        private String namespace;
+        private NameFormatter nameFormatter = DEFAULT_NAME_FORMATTER;
+        private String resourceGroup;
+        private Type[] scopes = getAllMetricScopes();
+        private boolean descriptionEnabled = true;
+        private int batchSize = DEFAULT_BATCH_SIZE;
+        private boolean enabled = true;
+        private Monitoring monitoringClient;
+
+        @Override
+        public OciMetricsSupport build() {
+            if (monitoringClient == null) {
+                throw new IllegalArgumentException("Monitoring client must be set in builder before building it");
+            }
+            return new OciMetricsSupport(this);
+        }
+
+        /**
+         * Sets the initial delay before metrics are sent to OCI
+         * (defaults to {@value #DEFAULT_SCHEDULER_INITIAL_DELAY}).
+         *
+         * @param value initial delay, expressed in time units set by {@link #schedulingTimeUnit(TimeUnit)}
+         * @return updated builder
+         */
+        @ConfiguredOption
+        public Builder initialDelay(long value) {
```

Comment:
The developer could invoke this and other builder methods directly, so they might need to have something like  `Objects.requireNonNull` invocations to check the parameter value before proceeding.

Reply:
Implemented on methods with non-primitive type parameters